connection.js

```diff
@@ -711,27 +729,27 @@ function get_r_u_arr_booleans(familyName, notificationRanked) {
     // for the booleans "only" checks, we are trying to find any reason to reject
     // the challenge, so the general and ranked/unranked args dont conflict.
     // (unlike --minmaintimeranked 50 --minmaintime 300)
+
     return [ [general,  true],
              [ranked,   notificationRanked],
              [unranked, !notificationRanked]
            ];
 }
 
 function getBooleans_r_u_Reject(argName, nameF, ending) {
-    const rankedUnranked = beforeRankedUnrankedGamesSpecial("for ", "", argName, "");
+    const rankedUnranked = getForFromBLCRankedUnrankedGames("for ", "", argName, "");
     const msg = `${nameF} not allowed on this bot ${rankedUnranked}${ending}.`;
     conn_log(msg);
     return { reject: true, msg };
 }
 
 function getBoardsizeNotSquareReject(argName, notificationWidth, notificationHeight) {
-    const rankedUnranked = beforeRankedUnrankedGamesSpecial("for ", "", argName, "");
+    const rankedUnranked = getForFromBLCRankedUnrankedGames("for ", "", argName, "");
     conn_log(`boardsize ${notificationWidth}x${notificationHeight} `
              + `is not square, not allowed ${rankedUnranked}`);
-    const msg = `Your selected board size ${notificationWidth}x${notificationHeight} `
-                + `is not square, not allowed ${rankedUnranked}, `
-                + `please choose a SQUARE board size (same width and `
-                + `height), for example try 9x9 or 19x19}`;
+    const msg = `Board size ${notificationWidth}x${notificationHeight} is not square`
```

Comment:
oh i never considered this but yes indeed, we technically do the same!

it would also make code simpler if in the future we want to allow non square boardsizes, just that the list will be long but we can find a fix for that too in the future :)

so we would just need to change reject messages in checkChallenge.test.js!
great idea!!! :100:

Reply:
actually i tried to do it but if we want to do that we should create a new function that checks allowed boardsizes specifically, because the checks are different (notif width, notif height, possibly support non square boardsizes, etc.)

something like this:

```Shell
checkAllowedBoardsizes -> width, height, etc.
checkAllowedFamily(komi.)
checkAllowedFamily(speed)
checkAllowedFamily(timecontrol)(modifié)
```
boardsize is too different from other allowed families, it should have its own separate function and just can't be together with the other ones

i'll open a github issue to keep that in min